src/components/Maps/MapPage.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { useQuery } from 'react-apollo';
+import GoogleMapReact from 'google-map-react';
+
+import MenuAppBar from '../common/MenuAppBar';
+import InfoWindow from './InfoWindow';
+import { LocationPin, GET_PINS_BY_EVENT_ID } from '../../graphql/queries/maps';
+
+const MapPage = ({
+  match: {
+    params: { eventId },
+  },
+}: {
+  match: { params: { eventId: string } };
+}) => {
+  const defaultMap = {
+    center: { lat: 43.470846, lng: -80.538473 },
+    zoom: 11,
+  };
+
+  const getMapOptions = (maps) => {
+    return {
+      streetViewControl: false,
+      scaleControl: true,
+      fullscreenControl: false,
+      styles: [
+        {
+          featureType: 'poi.business',
+          elementType: 'labels',
+          stylers: [
+            {
+              visibility: 'off',
+            },
+          ],
+        },
+      ],
+      gestureHandling: 'greedy',
+      disableDoubleClickZoom: true,
+
+      mapTypeControl: true,
+      mapTypeId: maps.MapTypeId.ROADMAP,
+      mapTypeControlOptions: {
+        style: maps.MapTypeControlStyle.HORIZONTAL_BAR,
+        position: maps.ControlPosition.BOTTOM_CENTER,
+        mapTypeIds: [
+          maps.MapTypeId.ROADMAP,
+          maps.MapTypeId.SATELLITE,
+          maps.MapTypeId.HYBRID,
+        ],
+      },
+
+      zoomControl: true,
+      clickableIcons: false,
+    };
+  };
+
+  const { data, loading } = useQuery(GET_PINS_BY_EVENT_ID, {
+    variables: { eventId },
+  });
+
+  const pins: Array<LocationPin> = data && !loading ? data.pinsForEvent : [];
+  const [infoWindowOpen, setInfoWindowOpen] = React.useState(false);
+  const [interestPinTitle, setInterestPinTitle] = React.useState('');
+  const [interestPinLocation, setInterestPinLocation] = React.useState('');
+
+  const initMaps = (map, maps) => {
+    if (!loading) {
+      map.addListener('click', () => {
+        setInfoWindowOpen(false);
+        setInterestPinLocation('');
+        setInterestPinTitle('');
+      });
+      pins.map((pin) => {
+        const marker = new maps.Marker({
+          position: { lat: pin.latitude, lng: pin.longitude },
+          map,
+          title: pin.label,
+        });
+
+        marker.addListener('click', () => {
+          map.setCenter(marker.getPosition());
+          setInfoWindowOpen(true);
+          setInterestPinLocation(pin.address);
+          setInterestPinTitle(pin.label);
+        });
+
+        return marker;
+      });
+
+      if (navigator.geolocation) {
```

Comment:
Separate marker for current location (which is why I have a `setIcon` call to distinguish this marker from other markers).

Reply:
tested out the current location and it was pretty off for me (maybe a km or 2 - dunno if it's just me though)

Reply:
Mine is off as well but not by that large of a margin, more like a 100-200m. [This](https://developers.google.com/maps/documentation/javascript/examples/map-geolocation) is the recommended way of getting user's current location on Google Maps though. There is an [older version](https://developers.google.com/maps/documentation/geolocation/overview) of geolocation but because it's older, I wasn't sure if we should use it.

Reply:
my second try was right on the money, likely not a big issue